### PR TITLE
function to comparisons variables and tests

### PR DESCRIPTION
**Pull request by LamAdr**
Tests translated from https://github.com/vincentarelbundock/marginaleffects/blob/main/inst/tinytest/test-comparisons.R

The option (from R's [issue #527](https://github.com/vincentarelbundock/marginaleffects/issues/527)) to pass a function to `comparisons`' `variables` argument seemed not implemented, and I figured it wouldn't be too hard to add. But maybe there's a reason it wasn't there yet? Should we be more cautious handling the output of the custom function?

Thanks

### CODE REVIEW

File: marginaleffects/sanity.py [hunk 1]
```diff
@@ -262,6 +262,12 @@ def clean(k):
             hi = newdata[variable] + value / 2
             lo = newdata[variable] - value / 2
 
+        elif callable(value):
+            tmp = value(newdata[variable])
```
Comment (vincentarelbundock) [inline marginaleffects/sanity.py:266]: Can you insert a check to make sure that `value` returns a data frame with two columns? If it doesn't, return an informative error.

**Review by vincentarelbundock — COMMENTED** [review]
Looks fantastic! I just added one request.

**Review by LamAdr — COMMENTED** [review]
@vincentarelbundock

**Comment by vincentarelbundock** [comment]
Thanks!

**Pull request merged**